pulsar/error.go

```diff
@@ -225,3 +226,14 @@ func getResultStr(r Result) string {
 		return fmt.Sprintf("Result(%d)", r)
 	}
 }
+
+func getErrorFromServerError(serverError *proto.ServerError) error {
```

Comment:
Maybe we can replace this function with a simple call like:

```go
newError(Result(*serverError), serverError.String())
```

Because `Result` and `proto.ServerError` should be just the same.

Reply:
Sorry I might give a wrong suggestion. I just reviewed the `Result` enum again, the `ServerError` cannot be converted to the `Error`. You have adjusted the order to make `InvalidStatus` (`Result`) equal to `ServerError_InvalidTxnStatus` (`ServerError`). However, other error codes might not be the same. So it's better to map `ServerError` to `Result` manually here, like what you did before.

Reply:
@liangyepianzhou Please also take a look at this comment. I've thought `Result` and `proto.ServerError` should be just the same before, but it seems that they are already different currently.

Reply:
Done